operator-pipeline-images/operatorcert/entrypoints/static_tests.py

```diff
@@ -77,13 +80,25 @@ def check_bundle(
     """
     repo = Repo(repo_path)
     operator = repo.operator(operator_name)
-    bundle = operator.bundle(bundle_version)
+    bundle = operator.bundle(bundle_version) if bundle_version else None
+    catalogs = OperatorCatalogList(
+        [
+            catalog
+            for catalog in operator.all_operator_catalogs()
+            if catalog in affected_catalogs.split(",")
+        ]
+    )
 
     outputs = []
     passed = True
 
     for suite_name in suite_names:
-        for result in run_suite([bundle, operator], suite_name, skip_tests=skip_tests):
+        for result in run_suite(
+            # do only catalog checks if bundle is not provided
+            [bundle, operator] if bundle else [catalogs],
```

Comment:
Why do we make it mutually exclusive? Can we run all checks if the bundle, operator, and catalog are available?

Reply:
As of now we have validation preventing simultaneous bundle and catalog changes, so we don't have both at the same time. And for current workflow, operator check was also skipped if bundle was not affected.
If it makes sense, to validate at least the operator together with the catalog, I can change the line to something like `[bundle, operator] if bundle else [catalogs, operator]` to have operator covered in both workflows.

Reply:
Changed so Operator checks are run in both cases.
As it is now, the `run_suite()` expects specific class instances and providing `None` instead of `Bundle` leads to an exception.

Reply:
Last update changes the logic to run each suite available.